flink-ml-core/src/main/java/org/apache/flink/ml/param/ParamValidators.java

```diff
@@ -38,6 +38,22 @@ public static <T> ParamValidator<T> gtEq(double lowerBound) {
         return (value) -> value != null && ((Number) value).doubleValue() >= lowerBound;
     }
 
+    // Checks if all values in the parameter intArray is greater than or equal to lowerBound.
+    public static <T> ParamValidator<T> intArrayGtEq(double lowerBound) {
+        return value -> {
```

Comment:
Given that we are making this a public method, I think it might be better to make it more generic to be useful in other algorithms as well. For example, instead of just validating integer arrays, we can make it validate any numeric arrays. Or we can introduce a `forEach` method, so that in order to validate indices, we can use `ParamValidators.forEach(ParamValidators.gtEq(0))`. What do you think of these ideas? we can discuss offline about this topic.

Reply:
I think making it more generic is better than introducing a forEach function. Introducing a forEach may make the user API more complicated. 

- ParamValidators.numericalArrayGtEq(0))
- ParamValidators.forEach(ParamValidators.gtEq(0))

Reply:
We agreed during an offline discussion that `ParamValidators.numericalArrayGtEq(0)` is a proper solution. For similar discussions we can directly refer to existing practices, like that in Alink or Spark.